shell-commands.nix

```diff
@@ -47,4 +54,6 @@ let
     };
   };
 in
-lib.makeOverridable inner allCommands
+(lib.makeOverridable inner allCommands).overrideAttrs (attrs: {
+  passthru.shellCommandHelpText = shellCommandHelpText attrs.passthru._descriptions;
```

Comment:
I'm thinking is that it could just be called `helpText` since the derivation already is shellCommands.

How is this intended to be used, because now it will be an string attribute with the bash code to print the help text, maybe it should be a derivation with the script in it instead?

Reply:
It can be printed in the `shellHook` which is the intended use case. Makes the shell command derivation easier to use for example with `mkShell` so it needs to be tied to the `shellCommands` derivation in some way (i.e. be a property on it). I agree on the name though.

Reply:
Another option would have been to use a setup hook to append itself to `shellHook` but that is a bit more invasive and not as flexible.